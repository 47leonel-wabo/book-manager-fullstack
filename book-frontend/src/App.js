import RestGithub from "./RestGithub";
import "./App.css";
import WeatherApp from "./WeatherApp";

function App() {
    return (
        <div className="App">
            <WeatherApp />
            <br />
            <RestGithub />
        </div>
    );
}

export default App;
