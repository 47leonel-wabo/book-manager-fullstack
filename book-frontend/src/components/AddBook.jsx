import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    TextField,
    Typography,
} from "@material-ui/core";
import React from "react";
import { useState } from "react";

const AddBook = ({ handleSave }) => {
    const [isOpen, setIsOpen] = useState(false);
    const [book, setBook] = useState({
        title: "",
        publisher: "",
        topic: "",
        isbn: "",
        publicationDate: "",
        price: "",
    });

    const handleClose = () => {
        setIsOpen(false);
    };

    const handleOpen = () => {
        setIsOpen(true);
    };

    const handleChange = (event) => {
        setBook({ ...book, [event.target.name]: event.target.value });
    };

    const handleSaveBook = () => {
        handleSave(book);
        handleClose();
    };

    return (
        <>
            <div>
                <button onClick={handleOpen}>Add book</button>
                <Dialog open={isOpen} onClose={handleClose}>
                    <DialogTitle>
                        <Typography variant="h6" color="inherit">
                            New Book
                        </Typography>
                    </DialogTitle>
                    <DialogContent>
                        <TextField
                            id="outline-basic"
                            variant="outlined"
                            size="small"
                            label="Book title"
                            fullWidth
                            margin="normal"
                            defaultValue={book.title}
                            name="title"
                            onChange={handleChange}
                        />
                        <TextField
                            id="outline-basic"
                            variant="outlined"
                            label="Publisher"
                            size="small"
                            fullWidth
                            margin="normal"
                            defaultValue={book.publisher}
                            name="publisher"
                            onChange={handleChange}
                        />
                        <TextField
                            id="outline-basic"
                            variant="outlined"
                            label="Book topic"
                            defaultValue={book.topic}
                            size="small"
                            fullWidth
                            margin="normal"
                            name="topic"
                            onChange={handleChange}
                        />
                        <TextField
                            id="outline-basic"
                            variant="outlined"
                            label="Book ISBN"
                            defaultValue={book.isbn}
                            fullWidth
                            margin="normal"
                            size="small"
                            name="isbn"
                            onChange={handleChange}
                        />
                        <TextField
                            id="outline-basic"
                            variant="outlined"
                            label="Publication Date"
                            size="small"
                            fullWidth
                            margin="normal"
                            defaultValue={book.publicationDate}
                            name="publicationDate"
                            onChange={handleChange}
                        />
                        <TextField
                            id="outline-basic"
                            variant="outlined"
                            label="Book Price"
                            size="small"
                            defaultValue={book.price}
                            name="price"
                            fullWidth
                            margin="normal"
                            onChange={handleChange}
                        />
                    </DialogContent>
                    <DialogActions>
                        <Button
                            variant="contained"
                            color="secondary"
                            onClick={handleClose}
                        >
                            Cancel
                        </Button>
                        <Button
                            variant="contained"
                            color="primary"
                            onClick={handleSaveBook}
                        >
                            Save
                        </Button>
                    </DialogActions>
                </Dialog>
            </div>
        </>
    );
};

export default AddBook;
