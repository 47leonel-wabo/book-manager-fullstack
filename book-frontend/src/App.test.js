import { render, screen } from "@testing-library/react";
import App from "./App";
import TestRenderer from "react-test-renderer";
import AddBook from "./components/AddBook";

test("renders book manager", () => {
    render(<App />);
    const bookManagerTitle = screen.getByText(/Book Manager/i);
    expect(bookManagerTitle).toBeInTheDocument();
});

test("renders export in csv link", () => {
    render(<App />);
    const exportCsvLink = screen.getByText(/Export CSV/i);
    expect(exportCsvLink).toBeInTheDocument();
});

it("render a snapshot", () => {
    const tree = TestRenderer.create(<AddBook />).toJSON();
    expect(tree).toMatchSnapshot();
});
