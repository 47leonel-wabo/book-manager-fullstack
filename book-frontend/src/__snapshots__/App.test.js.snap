// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Add book
    </span>
  </button>
</div>
`;
